exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h

```diff
@@ -71,27 +89,43 @@ class OtlpGrpcExporter final : public opentelemetry::sdk::trace::SpanExporter
   bool Shutdown(
       std::chrono::microseconds timeout = (std::chrono::microseconds::max)()) noexcept override;
 
+  /**
+   * Get the Client object
+   *
+   * @return return binded gRPC client
+   */
+  const nostd::shared_ptr<OtlpGrpcClient> &GetClient() const noexcept;
+
 private:
   // The configuration options associated with this exporter.
   const OtlpGrpcExporterOptions options_;
 
-#ifdef ENABLE_ASYNC_EXPORT
-  std::shared_ptr<OtlpGrpcClient> client_;
-#endif
+  nostd::shared_ptr<OtlpGrpcClient> client_;
```

Comment:
why is this switched from std::shared_ptr to nostd::shared_ptr?

Reply:
We allow OTLP exporter factories create exporters with existed `OtlpGrpcClient`, in which APIs are exported.
Should we keep the ABI compatibility of exported APIs?

Reply:
So, I made a second pass at code review, looking more closely at `std` versus `nostd` in the OTLP exporter public header files.

For some classes, we have no choice:

* `nostd::span`
* `nostd::string_view`

because the std part may not exist.

For opentelemetry objects returned by factory::Create() methods, all existing methods:

* do return a unique_ptr instead of a shared_pointer, because a factory just creates an instance, and does not know if it will be used later in a shared object or not
* do use `std::unique_ptr`

For the API, we provide strong(er) guarantees about compatibility, and use `nostd` all over the place.

For the SDK and Exporters, we do not provide such strong guarantees of ABI compatibility of the SDK/exporter interface to the application, which is why `std` classes are used already.

For example `OtlpGrpcExporterFactory::Create()` returns a `std::unique_ptr<opentelemetry::sdk::trace::SpanExporter>`, and this is part of the exporter ABI already.

Instead of having a mix of `std` and `nostd` for unique_ptr and shared_ptr, which is very inconvenient and a source of issues, I think it makes sense to use `std::unique_ptr` and `std::shared_ptr` in the exporter exposed interface, for OTLP GRPC.

@owent 

Having `std::shared_ptr<OtlpGrpcClient>` is just fine here, could you adjust the patch to use `std` classes everywhere, for `std::unique_ptr` and `std::shared_ptr` ?

Also, please change:

* OtlpGrpcClientFactory::Create()
* OtlpGrpcClientFactory::CreateReferenceGuard()

to return a `std::unique_ptr`.
The caller can then make a `std::shared_ptr` with it, and the rest of the code will use a shared_ptr like you have currently.

Reply:
OK, I will change them to `std::unique_ptr` and `std::shared_ptr` later.

Reply:
Done